lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -64,38 +64,77 @@ private function loadAttributes( $userId ) {
 	 * @param string $userId
 	 * @param Attribute $attribute
 	 */
-	public function setAttribute( $userId, $attribute ) {
+	public function setAttribute( $userId, Attribute $attribute ) {
 		if ( $this->isAnonUser( $userId ) ) {
 			return;
 		}
 
-		$this->loadAttributes( $userId );
-		$this->setAttributeInService( $userId, $attribute );
-		$this->setAttributeInCache( $userId, $attribute );
+		if ( $this->attributeValueAlreadySet( $userId, $attribute ) ) {
+			return;
+		}
 
+		$this->setAttributeInCache( $userId, $attribute );
+		$this->setAttributeInService( $userId, $attribute );
 	}
 
 	private function isAnonUser( $userId ) {
 		return $userId === 0;
 	}
 
+	private function attributeValueAlreadySet( $userId, Attribute $attribute ) {
+		return $this->getAttribute( $userId, $attribute->getName() ) == $attribute->getValue();
```

Comment:
Are there any "gotchas" here where an attribute would need to be set to `0` and this would return true if compared to `null`? In other words, should you use `===` here or are you sure it'll never be needed?